### ISSUE TITLE
Are different wallet versions fully compatible for sending TON?

### ISSUE THREAD

**Issue by akegaviar** [akegaviar/ton-foo#412]
There are several different wallet contract versions on TON, with the latest wallet contract version currently being v4r2. They have different abilities and they even produce different wallet addresses with the same mnemonic. So I'd like to be sure, are they fully compatible in sending and receiving Toncoins? Is it fully safe to send money from a wallet with one version to a wallet with another?

This question was imported from Telegram Chat: [https://t.me/tondev/127543](https://t.me/tondev/127543)

## Answer

Yes, that's correct.

You can take a look at the document here: [https://docs.ton.org/participate/wallets/contracts](https://docs.ton.org/participate/wallets/contracts)

Basically, the difference in wallets lies in whether it's a high-load wallet, whether it supports NFTs, subscriptions, etc.

---

[Original](https://answers.ton.org/question/1585283961360224256/are-different-wallet-versions-fully-compatible-for-sending-ton)
